tests/integration/test_integration.py

```diff
@@ -223,15 +223,16 @@ async def test_given_upf_machine_charm_is_deployed_and_in_blocked_status_when_ch
 
         assert application.status == "active"
 
-    @pytest.mark.abort_on_fail
-    async def test_given_grafana_agent_deployed_when_relate_to_grafana_agent_then_status_is_active(self, deploy_grafana_agent):  # noqa: E501
-        application = await self._get_application(APP_NAME)
-        await self.model.integrate(
-            relation1=f"{APP_NAME}:cos-agent",
-            relation2=f"{GRAFANA_AGENT_APPLICATION_NAME}:cos-agent",
-        )
-        await self.model.wait_for_idle(apps=[APP_NAME])
-        assert application.status == "active"
+    # FIXME: grafana-agent doesn't have a version working with Ubuntu 24.04
+    # @pytest.mark.abort_on_fail
+    # async def test_given_grafana_agent_deployed_when_relate_to_grafana_agent_then_status_is_active(self, deploy_grafana_agent):  # noqa: E501
+    #     application = await self._get_application(APP_NAME)
+    #     await self.model.integrate(
+    #         relation1=f"{APP_NAME}:cos-agent",
+    #         relation2=f"{GRAFANA_AGENT_APPLICATION_NAME}:cos-agent",
+    #     )
+    #     await self.model.wait_for_idle(apps=[APP_NAME])
+    #     assert application.status == "active"
```

Comment:
This test kills the entire run, because Juju tries to create a 22.04 machine on demand and add it to the model. New machine goes to pending state which breaks the entire model.

Reply:
Oh, ok then, let's keep this as-is.